db/import.go

```diff
@@ -187,6 +189,9 @@ func (db *Database) importDoc(docid string, body Body, isDelete bool, existingDo
 // migration if _sync property exists.  If _sync property is not found, returns doc and sets requiresImport to true
 func (db *Database) migrateMetadata(docid string, body Body, existingDoc *sgbucket.BucketDocument) (docOut *document, requiresImport bool, err error) {
 
+	// TODO: deal with reloading scenarios
```

Comment:
This is probably worth discussing.  With feed based migration, should it be reloading the expiry from the couchbase doc if there is a cas failure?  As it stands, it will be using the expiry from the DCP feed event, which might be stale.

Reply:
Looking into this more, it looks like the reloading behavior is only relevant for on-demand imports:

From `db.importDoc()`:

```
	docOut, _, err = db.updateAndReturnDoc(docid, true, existingDoc.Expiry, existingDoc, func(doc *document) (Body, AttachmentData, error) {

		// Perform cas mismatch check first, as we want to identify cas mismatch before triggering migrate handling.
		// If there's a cas mismatch, the doc has been updated since the version that triggered the import.  Handling depends on import mode.
		if doc.Cas != existingDoc.Cas {
			// If this is a feed import, cancel on cas failure (doc has been updated )
			if mode == ImportFromFeed {
				return nil, nil, base.ErrImportCasFailure
			}
```

Reply:
That's definitely the case for feed-based import - we don't retry on a CAS failure, since we can just wait for the update that caused the CAS mismatch to show up over DCP.  

I'm less sure about what's done for feed-based migration, though.

Reply:
After setting a breakpoint in `migrateMetadata()` and looking at the stack, it looks like the `mode` variable is 0 and should behave the same as the feed based import case.

![screen shot 2017-11-29 at 2 58 20 pm](https://user-images.githubusercontent.com/296876/33403638-ff1d76a2-d515-11e7-93d4-20fdff3e6bac.png)

Reply:
Makes sense for the feed-based import case.  For on-demand import, need to confirm that the existing retry handling (in importDoc) already covers CAS failures that happen in migrateMetadata.